### ISSUE TITLE
tealdbg / algod interaction --- do not block, just log

### ISSUE THREAD

**Issue by jeapostrophe** [algorand/go-algorand#1573]
Currently, when you use the WebDebuggerHook with tealdbg, then algod will block while the debugger process runs. This is certainly useful and powerful. But it is implausible to use when I am debugging the 8th transaction in a sequence of 20 transactions.

What I really want is just a way to collect the log of the transaction run annotated with the values at each step, like a dry-run does. (You could read this as a request to make it so there's an option to produce dry-run output on a live transaction)

But, I expect it would be relatively easy to have tealdbg retain all of the debugging steps internally, allow algod to progress, and then just use the web/cdt user interface to allow the user to browse what happened on the run (i.e. by single-stepping through it) but just turning off the ability to modify anything.

**Comment by jeapostrophe** [comment]
If someone else has this problem and wants something similar, I wrote a simple HTTP server that just logs all of the messages that the WebDebuggerHook sends it --- https://gist.github.com/5ddbb1fcd06b532c627445b704bee6d4 --- and immediately replies with an OK.

The JSON objects are fairly readable, especially if you have `data/transactions/logic/debugger.go` open --- https://github.com/algorand/go-algorand/blob/423e9592367b91bf6fc1cfc177b7ac8c3f24f312/data/transactions/logic/debugger.go --- to read the names. The main thing you want to remember is that things that are zero won't be included, so if a stack entry reads `{tt: 2}`, then that means it is a type 2, which is integer, and the value is 0.

**Comment by algorandskiy** [comment]
@jeapostrophe Are you talking about something like "tealdbg remote --trace" for printing out data from `DebugState`? Makes sense for me.

**Comment by jeapostrophe** [comment]
Yup; in my debug server --- https://github.com/reach-sh/reach-lang/blob/master/scripts/algorand-devnet/server.rkt --- I just create a directory of files, one for each `/exec/x` `POST`, and then just browse through them, reading them with my eye balls :)

The key thing here is that the application I'm testing runs dozens and dozens of transactions, and it is too onerous to step through ever single one OR capture the state and replay just the transaction I care about.

**Comment by yaovi-a** [comment]
@algorandskiy , Can you take a look at this and let us known where to move it?

**Comment by algorandskiy** [comment]
This is a small task, easy to do.

**Comment by jeapostrophe** [comment]
FWIW, I now use this patch --- https://github.com/reach-sh/reach-lang/blob/master/scripts/algorand-devnet/reach.patch --- which adds a new DirDebugger mode that is totally satisfactory for my needs. I think accepting it would be easier than doing this issue (which is quite old now) :)
